src/main/java/picard/illumina/ClusterDataToSamConverter.java

```diff
@@ -178,10 +179,10 @@ private SAMRecord createSamRecord(final ReadData readData, final String readName
 
         if (!molecularIndexes.isEmpty()) {
             if (!this.molecularIndexTag.isEmpty()) {
-                sam.setAttribute(this.molecularIndexTag, String.join("", molecularIndexes));
+                sam.setAttribute(this.molecularIndexTag, String.join(molecularIndexDelimiter, molecularIndexes));
             }
             if (!this.molecularIndexQualityTag.isEmpty()) {
-                sam.setAttribute(this.molecularIndexQualityTag, String.join("", molecularIndexQualities));
+                sam.setAttribute(this.molecularIndexQualityTag, String.join(molecularIndexDelimiter, molecularIndexQualities));
```

Comment:
I really want the lengths of the strings to be equal if possible, including the delimiter.